freqtrade/optimize/hyperopt/hyperopt_optimizer.py

```diff
@@ -136,7 +156,7 @@ def _get_params_dict(
 
         # Return a dict where the keys are the names of the dimensions
         # and the values are taken from the list of parameters.
-        return {d.name: v for d, v in zip(dimensions, raw_params, strict=False)}
+        return raw_params
```

Comment:
Is this method still necessary? It's not converting anything - it's just returning the same object again ... (and i don't think the missmatch can even happen).

Reply:
I guess it can be removed.
dimensions look like this: `dimensions: [CategoricalDistribution(['>', 'ca']), IntDistribution(low=15, high=25)` and 
raw_params looks like this: `raw_params: {'buy_bb_m_angle_op': 'ca', 'buy_bb_m_patterns': 15`
The reason for {d.name: v for d, v in zip(dimensions, raw_params, strict=False)} was to get something similar to how row_params looks now.

Reply:
_get_params_dict removed

Reply:
I tried to delete also the _get_params_dict, but there are tests patching this function in tests/optimize/test_hyperopt.py. 
```
    mocker.patch(
        "freqtrade.optimize.hyperopt.hyperopt_optimizer.HyperOptimizer._get_params_dict",
        return_value={"max_open_trades": -1},
    )
```

Reply:
will take a look at this - i'm pretty sure this can be "remocked" to not keep a pointless method around.

Reply:
worked ;)